README.md

```diff
@@ -32,6 +32,12 @@ And you can find the following:
  - [TODO.md](TODO.md) is a set of things to be worked on, if you'd like to contribute!
  - [Documentation](docs) is currently a place to document design, and eventually can be more user-facing docs.
 
+## Design
+
+ - A **MiniCluster** is an [indexed job](https://kubernetes.io/docs/tasks/job/indexed-parallel-processing-static/) so we can create N copies of the "same" base containers (each with flux, and the connected workers in our cluster)
+ - The flux config is written to a volume at `/etc/flux/config` (created via a config map) as a brokers.toml file.
+ - We use an [initContainer](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/) with an Empty volume (shared between init and worker) to generate the curve certificates (`/mnt/curve/curve.cert`). The broker sees them via the definition of that path in the broker.toml in our config directory mentioned above.
```

Comment:
yep - I almost commented that. What is currently happening is technically incorrect, but I think OK for this early development. What is actually happening:

1. there is one initContainer per pod, and since it has a shared volume with it's (TBA) worker pod, it generates a certificate for it. I've confirmed this locally
2. Since the pods aren't actually networked yet, there is no issue

Right now I'm trying to get the core / basics worked out - and the customization of the /etc/hosts is next, and I really just needed the broker to be happy. I suspect once that is working, we will need to:

1. Update the empty volume to generate the certificate somewhere else
2. Have only the first flux worker (do we call it the main one?) generate the curve.cert
3. Create a config map for a volume that is shared between all workers
4. Move it there!

So TLDR we will need to clean up this logic after the entire thing is functioning. For now (development) it's probably OK. I made the comment about curve certificates (plural) because that is the current state (albeit it's wrong and will need to be changed).